src/main/java/io/github/fablabsmc/fablabs/api/fluidvolume/v1/volume/ImmutableFluidVolume.java

```diff
@@ -0,0 +1,55 @@
+package io.github.fablabsmc.fablabs.api.fluidvolume.v1.volume;
+
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.math.Fraction;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.volume.api.FluidContainer;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.volume.api.FluidVolume;
+
+import net.minecraft.fluid.Fluid;
+import net.minecraft.nbt.CompoundTag;
+
+/**
+ * a fluid volume who's fluid, contents, and NBT cannot change.
+ */
+public class ImmutableFluidVolume extends FluidVolume {
```

Comment:
I personally don't feel *super* comfy about the concept of having non-final instance classes, but I'd be more ok with this if `FluidVolume` had an `isMutable()` method that this could override and return false.

Reply:
Will do

Reply:
done